File: TangYuCreated/leetcode/C/简单题目/204.c
int countPrimes(int n) {
    int i;
    int j;
    int sum = 0;
    int state = 1;
    if (n == 0 || n == 1 || n == 2)     return 0;
    for (i = 2;i < n;i++)
    {
        for (j = 2;j < i;j++)
        {
            if (i % j == 0)
            {
                state = 0;
                break;
            }
        }

        if (state)  sum++;
    }
    return sum;

}

File: TangYuCreated/leetcode/Python/简单题目/204.py
import math
class Solution:
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 3:
            return 0
        primes = n * [True] #representing 0,1,2,...,n - 1
        primes[0] = primes[1] = False
        for i in range(int(len(primes) ** 0.5) + 1):
            if primes[i]:
                primes[i ** 2::i] = [False] * len(primes[i ** 2::i])
        return sum(primes)
